tests/kafkatest/tests/core/api_versions_check_test.py

```diff
@@ -0,0 +1,78 @@
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.tests.produce_consume_validate import ProduceConsumeValidateTest
+from kafkatest.utils import is_int
+from kafkatest.version import LATEST_0_9, LATEST_0_8_2, TRUNK, LATEST_0_10, KafkaVersion
+
+
+# Tests to check api versions check is performed correctly.
+class ApiVersionsCheckTest(ProduceConsumeValidateTest):
+
+    def __init__(self, test_context):
+        super(ApiVersionsCheckTest, self).__init__(test_context=test_context)
+
+    def setUp(self):
+        self.topic = "test_topic"
+        self.zk = ZookeeperService(self.test_context, num_nodes=1)
+            
+        self.zk.start()
+
+        # Producer and consumer
+        self.producer_throughput = 10000
+        self.num_producers = 1
+        self.num_consumers = 1
+        self.messages_per_producer = 1000
+
+    @parametrize(broker_version=str(TRUNK), should_fail=False)
+    @parametrize(broker_version=str(LATEST_0_10), should_fail=True)
+    @parametrize(broker_version=str(LATEST_0_9), should_fail=True)
+    @parametrize(broker_version=str(LATEST_0_8_2), should_fail=True)
+    def test_api_versions_check(self, broker_version, should_fail):
+
+        self.kafka = KafkaService(self.test_context, num_nodes=1, zk=self.zk, version=KafkaVersion(broker_version), topics={self.topic: {
+                                                                    "partitions": 1,
+                                                                    "replication-factor": 1,
+                                                                    'configs': {"min.insync.replicas": 1}}})
+        self.kafka.start()
+         
+        self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka,
+                                           self.topic, throughput=self.producer_throughput,
+                                           message_validator=is_int,
+                                           compression_types=["none"],
+                                           version=KafkaVersion(str(TRUNK)))
+
+        self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka,
+                                        self.topic, consumer_timeout_ms=30000, new_consumer=True,
+                                        message_validator=is_int, version=KafkaVersion(str(TRUNK)))
+
+        try:
+            self.run_produce_consume_validate(lambda: wait_until(
+                lambda: self.producer.each_produced_at_least(self.messages_per_producer) == True,
+                timeout_sec=120, backoff_sec=1,
+                err_msg="Producer did not produce all messages in reasonable amount of time"))
+            if should_fail:
+                raise Exception("Producer from version {} should fail to produce to broker version {}".format(KafkaVersion(str(TRUNK)), KafkaVersion(str(broker_version))))
```

Comment:
Should we not check that we get a better error for the case where `ApiVersions` is available (i.e. `0.10.0.x`)?